xbmc/utils/ProgressJob.cpp

```diff
@@ -180,10 +180,20 @@ void CProgressJob::MarkFinished()
   if (m_progress != NULL)
   {
     if (m_updateProgress)
+    {
       m_progress->MarkFinished();
+      //We don't own this pointer and it will be deleted after it's marked finished
+      //just set it to nullptr so we don't try to use it again
+      m_progress = nullptr;
+    }
   }
   else if (m_progressDialog != NULL && m_autoClose)
+  {
     m_progressDialog->Close();
+    //We don't own this pointer and it will be deleted after it's marked finished
+    //just set it to nullptr so we don't try to use it again
+    m_progressDialog = nullptr;
```

Comment:
Are you sure that this also fixes a crash? Because m_progressDialog can never be deleted manually, that's the job of the window manager. Setting this to `nullptr` results in later calls using `m_progressDialog` to segfault.